dask/array/overlap.py

```diff
@@ -474,21 +477,48 @@ def overlap(x, depth, boundary):
            [100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100],
            [100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100]])
     """
+    from .routines import aligned_coarsen_chunks
+
+    def coerce_depth_in_chunk(depth_values, chunks, tries=None):
+        """Ensure that chunk size is larger than depth"""
+        new_chunks = [c for c in chunks]
+        i = 0
+        if tries is None:
+            tries = {i: 0 for i in range(len(chunks))}
+        for d, c in zip(depth_values, chunks):
+            maxd = max(d) if isinstance(d, tuple) else d
+            if maxd > min(c):
+                if maxd > c[-1] and len(c) >= 2:
+                    # if the last chunk is the culprit, add it to prior
+                    new_chunks[i] = (*c[:-2], c[-2] + c[-1])
+                else:
+                    # otherwise coarsen all chunks
+                    new_chunks[i] = aligned_coarsen_chunks(chunks=c, multiple=maxd)
+                tries[i] += 1
+                if tries[i] > 3:
+                    raise ValueError(
+                        "The overlapping depth %d is larger than your\n"
+                        "smallest chunk size %d. Rechunk your array\n"
+                        "with a larger chunk size or a chunk size that\n"
+                        "more evenly divides the shape of your array." % (d, min(c))
+                    )
+                # try this again as long as we haven't tried too many times for this axis
+                new_chunks = [
+                    c for c in coerce_depth_in_chunk(depth_values, new_chunks, tries)
+                ]
+            i += 1
+        return tuple(new_chunks)
```

Comment:
I wonder if this might be simpler if we handle it on a per-axis basis.

```python
>>> ensure_minimum_chunksize(size=10, chunks=(20, 5, 100, 25, 5))
(25, 100, 30)
```

And then we could call this on each of the chunks axes

```python
new_chunks = tuple(ensure_minimum_chunksize(depth, cs) for depth.values(), chunks)
if new_chunks != chunks:
    x = x.rechunk(new_chunks)
```

This might require that we normalize things like depth ahead of time, but that normalization should probaby be separate from this rechunking process.

Reply:
Thank you for looking at this PR! I think it's a lot more straightforward now.